Hierarchy delivery

LEDThereBeLight opened the pull request:
Basic delivery support for hierarchies.

Changes the course outline on the delivery side to render the full hierarchy as a "table of contents" with links to each page or assessment. Also changes the previous/next buttons to go across container levels. A new public function was added to the `Numbering` module to support this, and existing functions were refactored to take the `resolver` as an argument.

bitbldr reviewed (changes requested):
Looks good for the most part, just two things:

1. I ran into an issue while testing this. After going through a course, then realizing my assessment had no questions I added a question, then Published and now trying to access that assessment in delivery, I get the following error:
```
[error] #PID<0.3373.0> running OliWeb.Endpoint (connection #PID<0.3368.0>, stream id 1) terminated
Server: cold-otter-97.loca.lt:80 (http)
Request: GET /course/4dde05e8ca1973bcca9bffc13e1548820eee93a3/page/graded_assessment_1
** (exit) an exception was raised:
    ** (ArithmeticError) bad argument in arithmetic expression
        (oli 0.1.0) lib/oli/delivery/page/page_context.ex:96: Oli.Delivery.Page.PageContext.determine_previous_next/2
        (oli 0.1.0) lib/oli/delivery/page/page_context.ex:57: Oli.Delivery.Page.PageContext.create_page_context/3
        (oli 0.1.0) lib/oli_web/controllers/page_delivery_controller.ex:42: OliWeb.PageDeliveryController.page/2
        (oli 0.1.0) lib/oli_web/controllers/page_delivery_controller.ex:1: OliWeb.PageDeliveryController.action/2
        (oli 0.1.0) lib/oli_web/controllers/page_delivery_controller.ex:1: OliWeb.PageDeliveryController.phoenix_controller_pipeline/2
        (phoenix 1.5.6) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (oli 0.1.0) lib/oli_web/endpoint.ex:1: OliWeb.Endpoint.plug_builder_call/2
        (oli 0.1.0) lib/plug/debugger.ex:132: OliWeb.Endpoint."call (overridable 3)"/2
        (oli 0.1.0) lib/oli_web/endpoint.ex:1: OliWeb.Endpoint.call/2
```
Not sure if this is related to these changes or not.

2. Now that we removed the mini outline from every page, we should reclaim the empty space that was taken up by it and move "<- Course Outline" link to be above the content. There is just a lot of empty space now that could be utilized by content
<img width="957" alt="Screen Shot 2020-11-04 at 2 36 40 PM" src="https://user-images.githubusercontent.com/6248894/98159785-2a215300-1eab-11eb-88c4-813907d105ee.png">

LEDThereBeLight commented:
Unfortunately I could not seem to reproduce the argument error -- I tried different combinations of graded pages with an activity in different orderings in the course outline and couldn't get it to trigger. But I added a `nil` check as a workaround in the previous/next page buttons in case the page cannot be found in the hierarchy for some reason.

I also removed the sidebar and put the "back to course outline" button at the top.
